### PR TITLE
javadoc v2 fixes

### PR DESCRIPTION
**Pull request by emlaver**
Javadoc updates and fixes during v2 sync-android shakedown.

### CODE REVIEW

**Comment by emlaver** [comment]
Javadoc needed? https://github.com/cloudant/sync-android/blob/1317231e9c7d70fa4032ac04623617c2b454796a/cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java#L56

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
     private final File location; // needed for close/delete
     private final File extensionsLocation; // for extensions: currently attachments and indexes
 
-    /* This map should only be accessed inside synchronized(openDatastores) blocks */
+    /* This map should only be accessed inside synchronized(openDocumentStores) blocks */
```
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java]: Looks like this is just called `documentStores` now.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java [hunk 2]
```diff
@@ -215,6 +215,12 @@ public Query query() {
         return query;
     }
 
+    /**
+     * <p>
+     * Closes the DocumentStore instance.
+     * </p>
+     *
```
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java]: This may not be the correct place, but I wonder if we should add some documentation around the expected open/close behaviour. For example, I think the only way to safely access the `DocumentStore` is for the open/close to be controlled from something like an application lifecycle where an instance is obtained at the start and closed at the end.
If not I think we end up in situations like this:
Thread 1: `ds1 = DocumentStore.getInstance(...)`
Thread 2: `ds2 = DocumentStore.getInstance(...)`
ds1 and ds2 are the same instance, now if thread 1 calls `close()` accesses on thread 2 will get `IllegalStateException` for the `DocumentStore` being closed.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java [hunk 3]
```diff
@@ -30,7 +30,7 @@
 
     /**
      * @param rev                   A {@code DocumentRevision} containing the information for a
-     *                              revision from a remote datastore.
+     *                              revision from a remote DocumentStore.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java]: "remote datastore" in this context is meant to mean a couch or cloudant instance (or anything which supports the couch replication protocol)

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java [hunk 4]
```diff
@@ -46,7 +46,7 @@
 
     /**
      *  Constructs a new CDTQQueryExecutor using the indexes in 'database' to index documents from
-     *  'datastore'.
+     *  'DocumentStore'.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java]: This comment is completely wrong, having been copied/pasted from the CDT code.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java [hunk 5]
```diff
@@ -59,7 +59,7 @@ public IndexUpdater(Database database, SQLDatabaseQueue queue) {
      *  These indexes are assumed to already exist.
      *
      *  @param indexes Map of indexes and their definitions.
-     *  @param database The local datastore
+     *  @param database The local DocumentStore
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java]: These are `Database` objects

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java [hunk 6]
```diff
@@ -55,7 +55,7 @@
 
     /**
      *  Constructs a new QueryExecutor using the indexes in 'database' to find documents from
-     *  'datastore'.
+     *  'DocumentStore'.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java]: Comment probably needs to be reworked - quotes imply that these are parameter names.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java [hunk 7]
```diff
@@ -66,7 +66,7 @@
      *  Execute the query passed using the selection of index definition provided.
      *
      *  The index definitions are presumed to already exist and be up to date for the
-     *  datastore and database passed to the constructor.
+     *  DocumentStore and database passed to the constructor.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java]: `Database` and its underlying `SQLDatabaseQueue` are passed to ctor.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java (hunk 3 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java]: Something like `revision from a remote Couch or Cloudant instance` should work? I just don't want to use datastore if it might cause confusion with the transition from datastore -> documentStore.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java (hunk 4 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java]: What about:
```
Constructs a new IndexUpdater using the specified database to index documents from the SQLDatabaseQueue queue.
```

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java (hunk 6 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java]: What about:
```
Constructs a new QueryExecutor using the indexes in database to find documents from the SQLDatabaseQueue queue.
```

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java (hunk 3 above)
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/documentstore/ForceInsertItem.java]: `remote server` I think is suitably generic

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java [hunk 8]
```diff
@@ -57,9 +57,10 @@
  * </p>
  *
  * <p>
- * In most cases, users are advised to open store instances and use them for the entire lifetime of
- * their application. This is more efficient than opening and closing instances for the same
- * underlying file system location multiple times throughout the lifetime of the application.
+ * In most cases, users are advised to obtain a single store instance via
+ * `{@link DocumentStore#getInstance}` and use them for the entire lifetime of their application.
```
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java]: them > it

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java [hunk 9]
```diff
@@ -20,16 +20,31 @@
 import java.util.Arrays;
 import java.util.List;
 
+/**
+ * Class to manage replication lifecycle events.
```
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: @brynh perhaps some input here?

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java [hunk 10]
```diff
@@ -20,16 +20,31 @@
...
+ */
 public class ReplicationPolicyManager {
 
     private final List<Replicator> replicators = new ArrayList<Replicator>();
     private final ReplicationListener replicationListener = new ReplicationListener();
 
+    /**
+     * Interface with callback methods for replication lifecycle events.
```
Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: Maybe something like:
>Interface with methods to receive callbacks for different termination states of the replications configured by a ReplicationPolicyManager.

@brynh ?

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java (hunk 6 above)
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java]: Something like:
```Constructs a new QueryExecutor
@param database the database containing the documents to query
@param queue the queue used for accessing the indexes database
```

and similar for the `IndexUpdater` comment

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java (hunk 9 above)
Comment (brynh) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: Maybe `Manages the coordination of starting/stopping all replicators used in a replication policy and providing monitoring of when replications have completed.`

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java (hunk 10 above)
Comment (brynh) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: I'd change it slightly to:

> Interface with methods to receive callbacks for different termination states of the replications configured by a replication policy.

Note I haven't said `ReplicationPolicyManager` because this is now used by `ReplicationService` as well as `ReplicationPolicyManager` and on Android, you don't directly use `ReplicationPolicyManager`, so it could be a little confusing to the user.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java (hunk 8 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/documentstore/DocumentStore.java]: Fixed in a23743c.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java (hunk 9 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: Fixed in a23743c.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java (hunk 10 above)
Comment (emlaver) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: Fixed in a23743c.

Comment (ricellis) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: @brynh if the Android services don't use the manager, do you think the listener should be extracted into it's own class? Though we'll do that under another PR if so.

Comment (brynh) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicationPolicyManager.java]: @ricellis Android does use `ReplicationPolicyManager`, but it's hidden from the user (it's all done under the hood in ReplicationService), whereas on Java SE you'd use it directly. I guess it might make sense to put the listener in it's own class as it perhaps could be confusing as it is.

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java [hunk 11]
```diff
@@ -47,6 +47,8 @@
     /**
      *  Constructs a new IndexUpdater using the specified {@link Database} to index documents from
      *  the {@link SQLDatabaseQueue} queue.
+     *  @param database the {@link Database} containing the documents to query.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexUpdater.java]: Change introductory sentence to just
`Constructs a new IndexUpdater`

File: cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java [hunk 12]
```diff
@@ -56,6 +56,8 @@
     /**
      *  Constructs a new QueryExecutor using the indexes in {@link Database} to find documents from
      *  the {@link SQLDatabaseQueue} queue.
+     *  @param database the {@link Database} containing the documents to query.
```
Comment (tomblench) [inline cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/QueryExecutor.java]: Change introductory sentence to just
`Constructs a new QueryExecutor`
